### PR TITLE
Wistia support

### PR DESCRIPTION
**Pull request by edelgado**
Hi Pete,

Thanks for a great React-based player. This PR adds support for [Wistia](https://wistia.com) videos. I followed the structure from the YouTube component and I'll leave some specific notes in the diff below. 

I'd love to work with you to get Wistia support into a `react-player` 🙂

### CODE REVIEW

File: src/players/Wistia.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  getDuration () {
+    if (!this.isReady || !this.player || !this.player.duration) return
+    return this.player.duration()
+  }
+  getFractionPlayed () {
+    if (!this.isReady || !this.player || !this.player.percentWatched) return null
+    return this.player.percentWatched()
+  }
+  getFractionLoaded () {
+    return null
```
Comment (edelgado) [inline src/players/Wistia.js]: AFAIK, the Wistia JS player does not expose this info.

File: test/karma/ReactPlayer.js [hunk 2]
```diff
@@ -104,6 +105,14 @@ describe('ReactPlayer', () => {
     it('fires onDuration with delayed load', done => testDurationDelayed(TEST_VIMEO_URL, done))
   })
 
+  describe('Wistia', () => {
+    it('fires onStart', done => testStart(TEST_WISTIA_URL, done))
+    it('fires onPlay', done => testPlay(TEST_WISTIA_URL, done))
+    it('fires onDuration', done => testDuration(TEST_WISTIA_URL, done))
+    it('fires onDuration with delayed load', done => testDurationDelayed(TEST_WISTIA_URL, done))
+    it.skip('fires onPause', done => testPause(TEST_WISTIA_URL, done))
```
Comment (edelgado) [inline test/karma/ReactPlayer.js]: The `//fast.wistia.com/assets/external/E-v1.js` Wistia JS player is outputting some errors in Firefox. The player works as expected, but it makes the tests have some output. I'll reach out to the Wistia folks to see if they know what the issue may be.

**Comment by edelgado** [comment]
Hi @CookPete. I just merged master and added support for playback rate as well. Let me know what you think. I'd love to get Wistia support into the player.

**Comment by CookPete** [comment]
Thanks for the PR @edelgado. I've [tidied things up slightly](https://github.com/CookPete/react-player/commit/9c82bfd332076fabb5e549e6e2c97cc7d6666a03) but it was mainly just me being anal about code style and consistency of stuff. Published in `0.14.0`. Merry christmas! 🎄

**Comment by edelgado** [comment]
Thanks for the merge and cleanup @CookPete, Merry Christmas to you too :-)

**Pull request merged**